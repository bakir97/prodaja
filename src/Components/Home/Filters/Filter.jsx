import React from "react";
import { TextField, InputAdornment, Grid } from "@material-ui/core";
const style = {
  start: { paddingLeft: 5, paddingTop: 10, textAlign: "start" },
  end: { paddingRight: 5, paddingTop: 10, textAlign: "end" }
};
const sameAtt = {
  type: "number",
  variant: "outlined",
  InputProps: {
    endAdornment: <InputAdornment position="end">KM</InputAdornment>
  }
};
export default ({ position, ...other }) => {
  return (
    <Grid item xs={6} style={position === "start" ? style.start : style.end}>
      <TextField {...other} {...sameAtt} />
    </Grid>
  );
};
