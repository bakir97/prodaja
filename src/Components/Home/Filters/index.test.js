import Filter from "./index";
describe("Filter index", () => {
  const filters = jest.fn();
  afterEach(() => {
    filters.mockClear();
  });
  const wrapper = shallow(<Filter filters={filters} />);
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should call minCijena", () => {
    wrapper.instance().changeCijenaMin({ target: { value: "testmin" } });
    expect(filters).toBeCalledWith({ minCijena: "testmin" });
  });
  it("should call maxCijena", () => {
    wrapper.instance().changeCijenaMax({ target: { value: "testmax" } });
    expect(filters).toBeCalledWith({ maxCijena: "testmax" });
  });
  it("should call minTezina", () => {
    wrapper.instance().changeTezinaMin({ target: { value: "testmin" } });
    expect(filters).toBeCalledWith({ minTezina: "testmin" });
  });
  it("should call maxTezina", () => {
    wrapper.instance().changeTezinaMax({ target: { value: "testmax" } });
    expect(filters).toBeCalledWith({ maxTezina: "testmax" });
  });
});
