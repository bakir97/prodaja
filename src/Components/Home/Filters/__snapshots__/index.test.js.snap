// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter index should render 1`] = `
<WithStyles(Grid)
  alignContent="center"
  alignItems="center"
  container={true}
  justify="center"
  style={
    Object {
      "padding": 10,
    }
  }
>
  <Filters
    changeCijenaMax={[Function]}
    changeCijenaMin={[Function]}
    changeTezinaMax={[Function]}
    changeTezinaMin={[Function]}
  />
</WithStyles(Grid)>
`;
